codalab/rest/legacy.py

```diff
@@ -447,6 +469,23 @@ def get_worksheet_content(uuid):
     service = BundleService()
     return service.full_worksheet(uuid)
 
+# Needs to store the last unfinished bundle uuids because if a bundle just finished running,
+# unfinished_bundle_uuids won't include that bundle (from the backend points of view, it's a finished bundle)
+# whereas the frontend is still waiting to update its state from running to ready/failed.
+last_unfinished_bundle_uuids = set()
```

Comment:
We should avoid global state like this.  What happens if there are many frontends calling this method...?  The frontend should pass in the set of bundles it think are unfinished.